File: react-alicante-24/code/src/ai.ts
let article = ""

import { openai } from "@ai-sdk/openai";
import { generateText } from "ai";


async function main() {
const { text } = await generateText({
model: openai('gpt-4o'),
system:
'You are a professional writer. ' +
'You write simple, clear, and concise content.',
prompt: `Summarize the following article in 3-5 sentences: ${article}`,
});
console.log(text)
}

File: react-alicante-24/code/src/ai.ml
type model = [ `OpenAi of Open_ai.t | `Anthropic of Anthropic.t]

type options = { model : model; system : string; prompt : string }

type generate_text_result = { text : string }


external generateText : options -> generate_text_result Js.Promise.t
  = "generateText"
[@@mel.module "ai"]
